app/Main.hs

```diff
@@ -148,20 +160,36 @@ main :: IO ()
 main = do
   env <- getEnvironment
   opts <- execParser (optionsInfo env)
+  manager <- getManager opts
 
-  eres <- runExceptT $ runReaderT vaultEnv (env, opts)
+  eres <- runExceptT $ runReaderT vaultEnv (ExecCtx env opts manager)
   case eres of
     Left err -> hPutStrLn stderr (vaultErrorLogMessage err)
     Right newEnv -> runCommand opts newEnv
 
+getManager :: Options -> IO Manager
+getManager opts =
+  let
+    managerSettings = if oConnectInsecure opts
+                      then defaultManagerSettings
```

Comment:
I'm confused by the logic here: Why are there both `oConnectInsecure` and `oNoValidateCerts`? And why are they mutually exclusive?
I would expect the former to imply the latter.

(Also, I think a `where` would read nicer here)

Reply:
`oConnectInsecure` means "connect over HTTP". If we don't connect over TLS, we want the default manager (which doesn't attempt to set up a TLS connection, so we don't need to have any TLS specific settings).

I'll add a comment about this